packages/react-composites/src/composites/CallComposite/components/buttons/Devices.tsx

```diff
@@ -20,6 +20,16 @@ export const Devices = (props: {
   disabled?: boolean;
 }): JSX.Element => {
   const devicesButtonProps = usePropsFor(DevicesButton);
+  /* @conditional-compile-remove(rooms) */
+  const permissions = _usePermissions();
+  /* @conditional-compile-remove(rooms) */
+  if (!permissions.microphoneButton) {
+    devicesButtonProps.microphones = [];
```

Comment:
Avoid mutating variables (mutating can cause hard to debug side effects as this could change the object for anyone that has a reference to it)
Instead create a new variable:
```suggestion
const augmentedDeviceButtonProps = {
  ...devicesButtonProps,
  microphones: !permissions.microphoneButton ? [] : devicesButtonProps.microphones,
  cameras: !permissions.cameraButton? [] : devicesButtonProps.cameras,
}
```

Reply:
^should be memoized also

Reply:
I agree with James here and I need to avoid mutating variables on my PR as well :)

Reply:
done